NGitLab/Models/GroupHookUpsert.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
+
+namespace NGitLab.Models;
+
+public class GroupHookUpsert
+{
+    [Required]
+    [JsonPropertyName("url")]
+    public Uri Url { get; set; }
+
+    [JsonPropertyName("push_events")]
+    public bool PushEvents { get; set; }
```

Comment:
According to the [GitLab documentation](https://docs.gitlab.com/ee/api/groups.html#edit-group-hook), `push_events` is not required. I would make nullable the properties that are not required. Otherwise the API call could inadvertently overwrite settings that ought not to be.